### PR TITLE
[test/plugin] immigrate grpc scenario

### PR DESCRIPTION
**Pull request by arugal**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#3583

### CODE REVIEW

File: test/plugin/archetypes/jvm/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -73,7 +73,6 @@
             <plugin>
                 <groupId>org.springframework.boot</groupId>
                 <artifactId>spring-boot-maven-plugin</artifactId>
-                <version>1.5.9.RELEASE</version>
```
Comment (arugal) [inline test/plugin/archetypes/jvm/src/main/resources/archetype-resources/pom.xml]: FYI @dmsolr

**Comment by wu-sheng** [comment]
Test fails. Please recheck.

**Comment by arugal** [comment]
```
io.grpc.StatusRuntimeException: INTERNAL: HTTP/2 error code: INTERNAL_ERROR
Received Goaway
```

Hi, BlockingStub has this error in  under 1.5.0, anyone?

**Comment by wu-sheng** [comment]
> Hi, BlockingStub has this error in under 1.5.0, anyone?

I am not. Look like a gRPC internal error?

**Comment by arugal** [comment]
> > Hi, BlockingStub has this error in under 1.5.0, anyone?
> 
> I am not. Look like a gRPC internal error?

It looks like it. BlockingStub is not included in the old scenario, but I saw BlockingStub on the example of the grpc-java 1.5.x branch.

**Comment by wu-sheng** [comment]
`BlockingStub` is useful, we are even using it. Even we are using gRPC 1.14, 1.5 is a very old version. Are you still testing blocking stub? I recommend you do.

**Comment by arugal** [comment]
I'm ready to run the example provided by gprc-java.

**Comment by wu-sheng** [comment]
error on gRPC 1.9 now.

File: test/plugin/scenarios/grpc-scenario/config/expectedData.yaml [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                 entryEndpointId: 0, refType: CrossProcess, parentSpanId: 2, parentTraceSegmentId: not null,
+                 parentServiceInstanceId: 1, networkAddress: '127.0.0.1:18080', entryEndpoint: /grpc-scenario/case/grpc-scenario,
+                 entryServiceInstanceId: 1}
+      - segmentId: not null
+        spans:
+          - operationName: Greeter.sayHello
+            operationId: 0
+            parentSpanId: -1
+            spanId: 0
+            spanLayer: Unknown
```
Comment (aderm) [inline test/plugin/scenarios/grpc-scenario/config/expectedData.yaml]: spanLayer is unknown?

File: test/plugin/scenarios/grpc-scenario/pom.xml [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    <build>
+        <finalName>grpc-scenario</finalName>
+        <extensions>
+            <extension>
+                <groupId>kr.motd.maven</groupId>
+                <artifactId>os-maven-plugin</artifactId>
+                <version>1.5.0.Final</version>
+            </extension>
+        </extensions>
```
Comment (aderm) [inline test/plugin/scenarios/grpc-scenario/pom.xml]: why need os-maven-plugin?

File: test/plugin/scenarios/grpc-scenario/src/main/java/org/apache/skywalking/apm/testcase/grpc/controller/CaseController.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        channel = ManagedChannelBuilder.forAddress(gprcProviderHost, grpcProviderPort).usePlaintext(true).build();
+        greeterStub = GreeterGrpc.newStub(ClientInterceptors.intercept(channel, new ConsumerInterceptor()));
+        greeterBlockingStub = GreeterBlockingGrpc.newBlockingStub(ClientInterceptors.intercept(channel, new ConsumerInterceptor()));
+    }
+
+    @RequestMapping("/grpc-scenario")
+    @ResponseBody
+    public String testcase() {
+        greetService();
+//        greetBlockingService();
```
Comment (aderm) [inline test/plugin/scenarios/grpc-scenario/src/main/java/org/apache/skywalking/apm/testcase/grpc/controller/CaseController.java]: comment can be deleted

File: test/plugin/scenarios/grpc-scenario/pom.xml [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                                <descriptor>src/main/assembly/assembly.xml</descriptor>
+                            </descriptors>
+                            <outputDirectory>./target/</outputDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.xolstice.maven.plugins</groupId>
+                <artifactId>protobuf-maven-plugin</artifactId>
```
Comment (aderm) [inline test/plugin/scenarios/grpc-scenario/pom.xml]: why need protobuf-maven-plugin?

File: test/plugin/scenarios/grpc-scenario/config/expectedData.yaml (hunk 2 above)
Comment (arugal) [reply test/plugin/scenarios/grpc-scenario/config/expectedData.yaml]: Good catch.

File: test/plugin/scenarios/grpc-scenario/src/main/java/org/apache/skywalking/apm/testcase/grpc/consumr/ConsumerInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+public class ConsumerInterceptor implements ClientInterceptor {
+
+    private Logger logger = LoggerFactory.getLogger(ConsumerInterceptor.class);
```
Comment (aderm) [inline test/plugin/scenarios/grpc-scenario/src/main/java/org/apache/skywalking/apm/testcase/grpc/consumr/ConsumerInterceptor.java]: this is slf4j, down is log4j2 ， Suggested unity

Comment (aderm) [inline test/plugin/scenarios/grpc-scenario/src/main/java/org/apache/skywalking/apm/testcase/grpc/consumr/ConsumerInterceptor.java]: this is slf4j, down is log4j2 ， Suggested unity

File: test/plugin/scenarios/grpc-scenario/pom.xml (hunk 5 above)
Comment (arugal) [reply test/plugin/scenarios/grpc-scenario/pom.xml]: Hi, this is the test case for GRPC, which seems to be needed. refre [generated-code](https://github.com/grpc/grpc-java#generated-code)

**Comment by arugal** [comment]
Since I have never found the reason why ```INTERNAL: HTTP/2 error code: INTERNAL_ERROR Received Goaway``` appears, the current support list does not include ```1.0.0-1.5.0```

File: test/plugin/scenarios/grpc-scenario/support-version.list [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+#1.4.0
+#1.3.1
+#1.3.0
+#1.2.0
+#1.1.2
+#1.1.1
+#1.0.3
+#1.0.2
+#1.0.1
+#1.0.0
```
Comment (wu-sheng) [inline test/plugin/scenarios/grpc-scenario/support-version.list]: We don't need comment if you don't support, remove them.

Comment (wu-sheng) [reply test/plugin/scenarios/grpc-scenario/support-version.list]: Those are very old, even we support part of is, like you said, blocking not support. It is OK to ignore it. I don't think someone is still using this version, even there is, we could simply say, please update the gRPC version.

File: test/plugin/scenarios/grpc-scenario/pom.xml (hunk 5 above)
Comment (aderm) [reply test/plugin/scenarios/grpc-scenario/pom.xml]: OK

**Comment by wu-sheng** [comment]
Any expected update about this?

**Comment by arugal** [comment]
> Any expected update about this?

I gave up, as explained in ```support-version.list```.

File: test/plugin/scenarios/grpc-scenario/pom.xml (hunk 3 above)
Comment (wu-sheng) [reply test/plugin/scenarios/grpc-scenario/pom.xml]: The gRPC required this, if I remember right.

Comment (arugal) [reply test/plugin/scenarios/grpc-scenario/pom.xml]: Is needed.
Refer [generated-code](https://github.com/grpc/grpc-java#generated-code)

**Pull request merged**